app/src/main/java/com.example.weatherapp/data/services/WeatherService.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.weatherapp.data.services
+
+import io.reactivex.rxjava3.core.Observable
+import com.example.weatherapp.data.model.ForecastModel
+import com.example.weatherapp.data.mapper.WeatherMapper
+import com.example.weatherapp.data.services.api.WeatherApiProvider
+import com.example.weatherapp.data.services.builder.WeatherRequestBuilder
+
+class WeatherService {
+
+    private val api = WeatherRequestBuilder()
+    private val mapper = WeatherMapper()
+
+    fun getForecast(city: String): Observable<MutableList<ForecastModel>> {
+        return Observable.create { subscriber ->
+            val callResponse =
+                api.buildService(WeatherApiProvider::class.java).getWeather(city, APPID, METRIC)
+            val response = callResponse.execute()
+            if (response.isSuccessful) {
+                response.body()?.list?.let { subscriber.onNext(mapper.transformList(it)) }
+            } else {
+                subscriber.onError(Throwable(response.message()))
+            }
+        }
+    }
+
+    companion object {
+        private const val APPID = "c3d5d3d6044a870b02965a19a6e26c95"
```

Comment:
put this APP_ID in a buildConfigField on the gradle pls

Reply:
APP_ID removed from companion and added into BuildConfigFile property on gradle.